src/controller/SetUpCodePairer.cpp

```diff
@@ -161,8 +161,35 @@ void SetUpCodePairer::OnDiscoveredDeviceOverBleError(void * appState, CHIP_ERROR
 #endif // CONFIG_NETWORK_LAYER_BLE
 
 #if CHIP_DEVICE_CONFIG_ENABLE_DNSSD
-void SetUpCodePairer::OnDiscoveredDevice(const Dnssd::DiscoveredNodeData & nodeData)
+
+bool SetUpCodePairer::NodeMatchesCurrentFilter(const Dnssd::DiscoveredNodeData & nodeData)
 {
+    switch (currentFilter.type)
+    {
+    case Dnssd::DiscoveryFilterType::kShort:
+        return (nodeData.longDiscriminator >> 8) == currentFilter.code;
```

Comment:
It looks error prone at the end of the day to have `nodeData.longDiscriminator >> 8` to represent the short discriminator. In a followup would it be possible to expose a `shortDiscriminator` field on the `DiscoveredNodeData` struct ?

Reply:
We could write something to do the calculation automatically, but that would be in a different class than the setupcodepairer. The dnssd txt record only has the long one. The spec specifically says to use that to calculate the short.